### ISSUE TITLE
SOAP envelopes encoding problem

### ISSUE THREAD

**Issue by philicious** [dreamfactorysoftware/dreamfactory#68]
I think I found another bug with SOAP services.

I put in as JSON in API Docs

```
{
    "sessionId": "c87c8whatever5a642fdb1b0cbc",
    "type": "simple",
    "set": "4",
    "sku": "foobar3000",
    "productData": {
    "categories":[19,21],
    "websites":[1],
    "name":"foobar 3000",
    "description":"foobar",
    "short_description":"foobar",
    "weight":"0",
    "status":"1",
    "visibility":"4",
    "price":"100",
    "tax_class_id":"2",
    "additional_attributes":{
      "single_data":[{
        "key":"manufacturer",
        "value":"232"
      }]
    }
  }
}
```

The JSON should be correct. I took a working PHP example and json_encoded the object. 

And the sniffed/dumped corresponding SOAP generated by DF is

```
<?xml version="1.0" encoding="UTF-8"?>
<SOAP-ENV:Envelope xmlns:SOAP-ENV="http://schemas.xmlsoap.org/soap/envelope/" xmlns:ns1="urn:Magento"><SOAP-ENV:Body>
<ns1:catalogProductCreateRequestParam>
<sessionId>c87c8whatever5a642fdb1b0cbc</sessionId>
<type>simple</type>
<set>4</set>
<sku>foobar3000</sku>
<productData>
    <categories><complexObjectArray>19</complexObjectArray><complexObjectArray>21</complexObjectArray></categories>
    <websites><complexObjectArray>1</complexObjectArray></websites>
    <name>foobar 3000</name>
    <description>foobar</description>
    <short_description>foobar</short_description>
    <weight>0</weight>
    <status>1</status>
    <visibility>4</visibility>
    <price>100</price>
    <tax_class_id>2</tax_class_id>
    <additional_attributes/>
</productData>
</ns1:catalogProductCreateRequestParam></SOAP-ENV:Body></SOAP-ENV:Envelope>
```

and as you can see the `additional_attributes`are empty.

**Comment by philicious** [comment]
@drewpearce any thoughts on this ? Its not blocking me but will likely affect someone else somewhen

**Comment by drewpearce** [comment]
@philicious I assume you tried this in the latest release?

**Comment by philicious** [comment]
@drewpearce  no, not yet. was with 2.1.1 but since there wasnt any update on this issue, I assume problem still exists. 

I yet have to update to 2.2 from 2.1.2 somewhen next days. So I cant try with latest-greatest right now.

**Comment by drewpearce** [comment]
I'm not sure if it's been addressed, but a lot of SOAP changes were made for 2.2 so it would be worth checking on. I'll have a chat with an engineer as well.

**Comment by philicious** [comment]
@drewpearce I just rechecked and this still applies for DF 2.2 with df-soap 0.3.0

**Comment by philicious** [comment]
any chance this will ever be addressed ? its still not working with DF 2.3 / df-soap 0.4.2

maybe @leehicks has an idea ?

**Comment by drewpearce** [comment]
This should be resolved now.
